elements/lisk-chain/src/state_store/state_store.ts

```diff
@@ -222,14 +224,25 @@ export class StateStore {
 	// createSnapshot follows the same interface as stateDB. However, it does not support multi snapshot.
 	public createSnapshot(): number {
 		this._snapshot = this._cache.copy();
-		return 0;
+		this._latestSnapshotId += 1;
+
+		if (this._latestSnapshotId > Number.MAX_SAFE_INTEGER) {
```

Comment:
```suggestion
		if (this._latestSnapshotId === Number.MAX_SAFE_INTEGER) {
```
At this point, the number is already exceeding the SAFE_INTEGER range. it's better to always keep it in the safe interger range